app/org/sagebionetworks/bridge/services/HealthDataService.java

```diff
@@ -156,4 +158,36 @@ public HealthDataAttachmentBuilder getAttachmentBuilder() {
     public HealthDataRecordBuilder getRecordBuilder() {
         return healthDataDao.getRecordBuilder();
     }
+
+    /**
+     * returns received list of record Ids after updating
+     * @param recordExportStatusRequest
+     *         POJO contains: a lit of health record ids, not upload ids and
+     *         an Synapse Exporter Status with value either NOT_EXPORTED or SUCCEEDED
+     * @return updated health record ids list
+     */
+    public List<String> updateRecordsWithExporterStatus(RecordExportStatusRequest recordExportStatusRequest) {
+        List<String> healthRecordIds = recordExportStatusRequest.getRecordIds();
+        HealthDataRecord.ExporterStatus synapseExporterStatus = recordExportStatusRequest.getSynapseExporterStatus();
+        if (recordExportStatusRequest.getRecordIds().isEmpty() || recordExportStatusRequest.getSynapseExporterStatus()== null) {
+            throw new InvalidEntityException(String.format(Validate.CANNOT_BE_BLANK, "healthRecordIds/synapseExporterStatus"));
+        }
```

Comment:
Assert that recordExportStatusRequest is not null. It'll fail immediately here but it's always helpful to be told in the error message what is wrong.

Check these fields separately and throw separate BadRequestExceptions, unless you are going to make this an entity. If you're going to make this an entity (and it probably should be, because it's in the API), you need to create a validator that throws InvalidEntityException with the property field-level errors.

The error below about the maximum number of records would still be separate and would still be a BadRequestException, because the entity is valid but the call didn't meet the preconditions of the API.